### DISCUSSION TITLE
async calls ?

### DISCUSSION THREAD

**Discussion by dignac3** [lkaybob/php-fcm-v1#18] [Ideas]
Hello there, I'm really happy to finally find a simple library ta call FCM V1 that work out of the box :100: !

I'm still curious about how you manage token lifecycle even if I saw the code (Symfony do it alone ? ou store it somewhere ?  ) 

I'm curious about one thing which is asynchronous call with Guzzle. It appears that making request to Google takes time & rise the response time. 
I would like to discuss & why not participate to improve your library with this kind of functionnality

**Comment by lkaybob** [comment]
Glad you found the place you want.

Before I begin my answer, I would like to point out that I didn't meant this package to be production-ready.
(I was merely trying out FCM's HTTP v1 API.)

> I'm still curious about how you manage token lifecycle even if I saw the code

Access token lifecycle is not considered, so it doesn't store access token anywhere. It contacts Google's OAuth server every time FCM message is sent.

Of course, it would be good to consider storing access token, since TTL of the acquired token is 1 hour as I remember.

> I'm curious about one thing which is asynchronous call with Guzzle. It appears that making request to Google takes time & rise the response time.

I'm not sure I haved used asynchronous call in my code. Could you point out where you seem to be?

> I would like to discuss & why not participate to improve your library with this kind of functionnality

As you could see from reposiotry's commit history, this package is written so many years ago. During that period, I shifted my role to other role rather than PHP developer. Therefore in reality, my continuous committment on the repository is limited.

Hope my reply helps answer your questions.
